modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -352,10 +353,29 @@ export class Transaction extends BaseTransaction {
     }
   }
 
+  constructSignedPayload(signature: Buffer): void {
+    const edSignature = `0x00${signature.toString('hex')}` as HexString;
+
+    this._signedTransaction = construct.signedTx(this._dotTransaction, edSignature, {
```

Comment:
Can we validate `edSignature` is valid before using it? Or will this throw if it isn't? If it throws, is the error descriptive enough to guide the user on what failed? As in, maybe we can capture the error and print the tx and signature if it is not printed already

Reply:
Good idea.
Not sure if we can validate it before we actually use it to sign. but instead I added a try catch to throw a more descriptive error message since the error it displayed wasn't descriptive enough. 

`ex: Unable to create Enum via index 124, in Ed25519, Sr25519, Ecdsa
`
Updated to this:

 ```
  try {
      this._signedTransaction = construct.signedTx(this._dotTransaction, edSignature, {
        registry: this._registry,
        metadataRpc: this._dotTransaction.metadataRpc,
      });
    } catch (e) {
      throw new SigningError(`Unable to sign dot transaction with signature ${edSignature} ` + e);
    }
```